types/validation.go

```diff
@@ -59,18 +59,8 @@ func VerifyCommit(chainID string, vals *ValidatorSet, blockID BlockID,
 			votingPowerNeeded, ignore, count, true)
 	}
 
-	// only count the signatures that are for the block
-	count := func(c CommitSig) bool { return c.BlockIDFlag == BlockIDFlagCommit }
-
-	// attempt to batch verify
-	if shouldBatchVerify(vals, commit) {
-		return verifyCommitBatch(chainID, vals, commit,
-			votingPowerNeeded, ignore, count, true, true)
-	}
-
-	// if verification failed or is not supported then fallback to single verification
-	return verifyCommitSingle(chainID, vals, commit, votingPowerNeeded,
-		ignore, count, true, true)
+	proposerKeyType := vals.GetProposer().PubKey.Type()
```

Comment:
This introduces an attack vector if the app allows registering a not-BLS validator. Say a validator joins with another key. If beacon-kit refuses to add it because key is not BLS, then it's okay. But if it contains a bug, the whole chain fails once it tries to verify commit where proposer is not BLS! It doesn't cost us anything to preserve the old code, so I don't see a point in panicking here.

Reply:
We can add an assert in "consensus paramas" validation to only contain BLS. That would block any validator having any other key type, and a clean way IMO